src/targets/gpu/convolution.cpp

```diff
@@ -121,6 +152,69 @@ shape miopen_convolution::find(context& ctx, const shape& output_shape, std::vec
         MIGRAPHX_THROW("MIOpen Convolution: find convolution failed");
     algo = perf.fwd_algo;
 
+    if(enabled(MIGRAPHX_MIOPEN_FIND_2{}))
+    {
+        miopenProblem_t conv_problem;
+        status = miopenCreateConvProblem(&conv_problem, cd.get(), miopenProblemDirectionForward);
+        if(status != miopenStatusSuccess)
+        {
+            MIGRAPHX_THROW("conv problem creation failed");
+        }
+        auto set_tensor_descriptor = [conv_problem](miopenTensorArgumentId_t name,
+                                                    tensor_descriptor& desc) {
+            auto status = miopenSetProblemTensorDescriptor(conv_problem, name, desc.get());
+            if(status != miopenStatusSuccess)
+            {
+                MIGRAPHX_THROW("setting problem tensor description failed");
+            }
+        };
+
+        set_tensor_descriptor(miopenTensorConvolutionX, x_desc);
+        set_tensor_descriptor(miopenTensorConvolutionW, w_desc);
+        set_tensor_descriptor(miopenTensorConvolutionY, y_desc);
+
+        auto solutions    = std::vector<miopenSolution_t>{};
+        std::size_t found = 0;
+        solutions.resize(100);
+
+        // TODO: create managed pointer for find_options
+        miopenFindOptions_t options;
+
+        // add checks for return statues  later
+        status = miopenCreateFindOptions(&options);
+        // not sure which tuning option to select
+        status = miopenSetFindOptionTuning(options, 1);
+        // not needed by default it is ordered by time
+        status = miopenSetFindOptionResultsOrder(options, miopenFindResultsOrderByTime);
+        // not sure if i should be using workspace size from GetWorkSpaceSize() call from earlier
+        status = miopenSetFindOptionWorkspaceLimit(options, workspace_size);
+
+        status = miopenFindSolutions(
+            miopen_handle, conv_problem, options, solutions.data(), &found, solutions.size());
+        // TODO: let manage pointer handle destruction
+        status = miopenDestroyFindOptions(options);
+        assert(found >= 0); // should it be throwing ?
+        solutions.resize(found);
+        // get the workspace size of the selected solution
+        auto solution = solutions.front();
+        status        = miopenGetSolutionWorkspaceSize(solution, &workspace_size);
+        // save solution
+        std::size_t solution_size;
+        status = miopenGetSolutionSize(solution, &solution_size);
+
+        auto solution_binary = std::vector<char>{};
+        solution_binary.resize(solution_size);
+        status          = miopenSaveSolution(solution, solution_binary.data());
+        solution_object = value::binary{solution_binary.data(), solution_size};
+
+        status = miopenDestroySolution(solution);
+        // which  of the workspace sizes it should return ? from GetSolutionWorkspaceSize or from
+        // ForwardGetWorkspaceSize
```

Comment:
It should return workspace from `GetSolutionWorkspaceSize`. `ForwardGetWorkspaceSize` returns the workspace needed to run Find(which will be the largest workspace for all solutions).

Reply:
Also, for find 2.0, you should need to call `ForwardGetWorkspaceSize` size either.

Reply:
One doesn't need `ForwardGetWorkspaceSize` for find 2.0. `miopenFindSolutions` doesn't require allocating anything from outside. After that point `miopenGetSolutionWorkspaceSize` would return the exact amount of workspace the solution would require to be executed. `ForwardGetWorkspaceSize` result may exceed what one needs to execute a specific solution. Using it implies that the backends behind find 2.0 and find 1.0 are the same, which may not be true in future.

Reply:
Yea thanks for the explanation. Sorry I was missing a 'not' in my comment above.

Reply:
Made changes,thanks.